static/js/ui_init.js

```diff
@@ -499,7 +499,11 @@ export function initialize_everything() {
     popover_menus.initialize();
 
     people.initialize(page_params.user_id, people_params);
-    settings_data.initialize(people.get_by_user_id(page_params.user_id).date_joined);
+
+    if (!page_params.is_spectator) {
+        const user = people.get_by_user_id(page_params.user_id);
+        settings_data.initialize(user.date_joined);
+    }
```

Comment:
I think we probably want to still initialize `settings_data`, just with now (`new Date()`?) for `date_joined`, for spectators.  I'd write this as an if/else statement to compute `date_joined` and then initialize `settings_data` unconditionally.